tests/platforms/kubernetes/appmanager.go

```diff
@@ -196,6 +202,53 @@ func (m *AppManager) ValidiateSideCar() (bool, error) {
 	return true, nil
 }
 
+// DoPortForwarding performs port forwarding to access test apps in the cluster
+func (m *AppManager) DoPortForwarding() error {
+	podClient := m.client.Pods(m.namespace)
+	// Filter only 'testapp=appName' labeled Pods
+	podList, err := podClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", TestAppLabelKey, m.app.AppName),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	var podName string
+
+	for _, pod := range podList.Items {
+		podName = pod.Name
+		break
+	}
+
+	config:= m.client.GetClientConfig()
+
+	// create spdy roundtripper
+	roundTripper, upgrader, err := spdy.RoundTripperFor(config)
+	if err != nil {
+		return err
+	}
+
+	// create stop and ready channels
+	stopChan, readyChan := make(chan struct{}, 1), make(chan struct{}, 1)
+	out, errOut := new(bytes.Buffer), new(bytes.Buffer)
+
+	path := fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", m.namespace, podName)
+	hostIP := strings.TrimLeft(config.Host, "https:/")
+	serverURL := url.URL{Scheme: "https", Path: path, Host: hostIP}
```

Comment:
If you are doing port forward, does it mean the certificate will not match the hostname anymore?